src/rlookup.c

```diff
@@ -393,9 +394,13 @@ int RLookup_LoadDocument(RLookup *it, RLookupRow *dst, RLookupLoadOptions *optio
   if (options->dmd) {
     dst->sv = options->dmd->sortVector;
   }
+  int ret;
+  LockHandler_AcquireGIL(options->sctx->redisCtx);
```

Comment:
The lock shouldn't be here; it should be enclosed elsewhere. Alternately, we should make sure that this stage is always executed on the main thread anyway... Especially if we are doing a kind of LOAD scenario, then this will be executed for every document.

Reply:
this is not happened in the main thread, I want to take the GIL only here cause here we read from the key space..

Reply:
We need to enforce this externally and not from within a function that can be called millions of times. This means that in the case of LOAD or similar (basically, if we need to read from the keyspace in the middle of processing the results), we should force the entire query to run on the main thread, since it would not make sense to run such queries in parallel anyway.

Note that we can determine from the query plan and rlookup which fields need to be loaded from the keyspace...